plugins/kubernetes/app/models/kubernetes/release.rb

```diff
@@ -25,11 +25,10 @@ def user
     def self.build_release_with_docs(params)
       roles = params.delete(:grouped_deploy_group_roles).to_a
       release = new(params) do |release|
-        if roles.flatten(1).any? { |dgr| dgr.kubernetes_role.blue_green? }
-          release.blue_green_color = begin
-            release.previous_succeeded_release&.blue_green_color == "blue" ? "green" : "blue"
-          end
-        end
+        # We always set the blue green color but the template filler might not always use it.
+        # This allows us to support deploy-specific features like env vars from config maps.
+        release.blue_green_color =
```

Comment:
this means we now always set the BLUE_GREEN env var ?
... did you check all other places that use blue_green_color to make sure it does not do anything silly ?

Reply:
Yeah, I need to do some digging, we should probably check the `blue_green?` flag rather than operate on the presence of a color.

Reply:
So, this isn't actually a problem. Because the template filler get's its `blue_green_color` from the release doc

https://github.com/zendesk/samson/blob/2951152daf3f001206f348f6267d7e6aaa9ef1ce/plugins/kubernetes/app/models/kubernetes/template_filler.rb#L570-L573

and the release doc guards the blue green color with the blue_green setting from the role. 

https://github.com/zendesk/samson/blob/2951152daf3f001206f348f6267d7e6aaa9ef1ce/plugins/kubernetes/app/models/kubernetes/release_doc.rb#L81-L83

similarly, the blue green env var pulls its value from the doc, which again is guarded by the role setting. 

https://github.com/zendesk/samson/blob/2951152daf3f001206f348f6267d7e6aaa9ef1ce/plugins/kubernetes/app/models/kubernetes/release_doc.rb#L125-L126

with that said, I would still be happy to define `blue_green?` on the release doc and guard the usage of the color with that instead of presence if we want. Not sure if we want that many changes though.

Reply:
guarding the use with `blue_green?` sounds good ... and then we use `blue_green` anyway for env configmaps